xml/System.Text.Json/JsonSerializerOptions.xml

```diff
@@ -1162,17 +1162,32 @@ Use <xref:System.Text.Json.Serialization.ReferenceHandler.Preserve> to enable un
+            or when deserializing a <see langword="null" /> value into a non-nullable property setter or constructor parameter.
+            Nullability annotations are resolved from the properties, fields, and constructor parameters
+            that are used by the serializer. This includes annotations stemming from attributes such as
+            <see cref="T:System.Diagnostics.CodeAnalysis.NotNullAttribute" />, <see cref="T:System.Diagnostics.CodeAnalysis.MaybeNullAttribute" />,
+            <see cref="T:System.Diagnostics.CodeAnalysis.AllowNullAttribute" /> and <see cref="T:System.Diagnostics.CodeAnalysis.DisallowNullAttribute" />.
+          </para>
           <para>
-            <see cref="T:System.Diagnostics.CodeAnalysis.AllowNullAttribute" /> and <see cref="T:System.Diagnostics.CodeAnalysis.DisallowNullAttribute" />.</para>
-          <para>Due to restrictions in how nullable reference types are represented at run time,</para>
-          <para>this setting only governs nullability annotations of non-generic properties and fields.</para>
-          <para>It cannot be used to enforce nullability annotations of root-level types or generic parameters.</para>
-          <para>The default setting for this property can be toggled application-wide using the</para>
-          <para>"System.Text.Json.Serialization.RespectNullableAnnotationsDefault" feature switch.</para>
+            Due to restrictions in how nullable reference types are represented at run time,
+            this setting only governs nullability annotations of non-generic properties, fields, and constructor parameters.
+            It cannot be used to enforce nullability annotations of root-level types, collection elements, or generic parameters.
+            The default setting for this property can be toggled application-wide using the
+            "System.Text.Json.Serialization.RespectNullableAnnotationsDefault" feature switch.
+          </para>
+          <para>
+            Nullability validation functions independently of required-ness validation, 
+            that is to say the setting has no effect in scenaria where the payload is missing a required property.
+            This can be configured independently using the <see langword="required" /> keyword, 
+            the <see cref="P:System.Text.Json.Serialization.JsonRequiredAttribute" /> attribute, 
+            or the <see cref="P:System.Text.Json.JsonSerializerOptions.RespectRequiredConstructorParameters" /> property.
+          </para>
+          <para>
+            It is recommended that new applications always set this property to <see langword="true" />,
+            in combination with the closely related <see cref="P:System.Text.Json.JsonSerializerOptions.RespectRequiredConstructorParameters" /> property.
+          </para>
         </remarks>
         <exception cref="T:System.InvalidOperationException">Thrown if this property is set after serialization or deserialization has occurred.</exception>
```

Comment:
I know it's not part of your current PR but could you also fix these exceptions? They need to read as if they started with "The exception that is thrown when...":

```suggestion
        <exception cref="T:System.InvalidOperationException">This property is set after serialization or deserialization has occurred.</exception>
```